Running "lein ring server" crashes with Clojure 1.5.0 onwards, requires lein-ring 0.8.0 or later

Thanks for this template!

...I found something odd. A newly generated `project.clj` will contain a dependency on Clojure 1.5.1 and lein-ring 0.7.5, but running `lein ring server` crashes with this stack trace:

```
$ lein ring server
Exception in thread "main" java.lang.AssertionError: Assert failed: (vector? (:dependencies project []))
at leinjacker.deps$add_if_missing.invoke(deps.clj:43)
at leiningen.ring.server$server_task.invoke(server.clj:22)
at leiningen.ring.server$server.invoke(server.clj:34)
at clojure.lang.AFn.applyToHelper(AFn.java:161)
at clojure.lang.AFn.applyTo(AFn.java:151)
at clojure.core$apply.invoke(core.clj:619)
at leiningen.ring$ring.doInvoke(ring.clj:22)
at clojure.lang.RestFn.invoke(RestFn.java:425)
at clojure.lang.Var.invoke(Var.java:419)
at clojure.lang.AFn.applyToHelper(AFn.java:163)
at clojure.lang.Var.applyTo(Var.java:532)
at clojure.core$apply.invoke(core.clj:619)
at leiningen.core.main$resolve_task$fn__2171.doInvoke(main.clj:149)
[... snip ...]
```

Tweaking the values on `project.clj` I found that it works with Clojure **1.4.0** instead (but not newer) and lein-ring with versions 0.8.0 through 0.8.5 (the most recent as of this writing).

With Clojure 1.5.0 and lein-ring 0.8.0 the stack trace changes to this instead:

```
$ lein ring server
Exception in thread "main" java.lang.IllegalArgumentException: contains? not supported on type: clojure.lang.LazySeq
```

So I think it's definitely something that changed recently...

BTW, I'm using this lein:

```
$ lein --version
Leiningen 2.1.2 on Java 1.7.0_21 OpenJDK Client VM
```

and upgrading to Lein 2.1.3 didn't help either.

Cheers!

--- a/src/leiningen/new/lein_webapp_template/project.clj
+++ b/src/leiningen/new/lein_webapp_template/project.clj
@@ -3,11 +3,11 @@
   :url "http://example.com/FIXME"
   :license {:name "Eclipse Public License"
             :url "http://www.eclipse.org/legal/epl-v10.html"}
-  :dependencies [[org.clojure/clojure "1.5.1"]
+  :dependencies [[org.clojure/clojure "1.4.0"]
                  [ring "1.1.6"]
                  [compojure "1.1.3"]
                  [stencil "0.3.2"]]
-  :plugins [[lein-ring "0.7.5"]]
+  :plugins [[lein-ring "0.8.5"]]
   :ring {:handler {{name}}.app/site-handler}
   :war-resources-path "resources/public"
   :main {{name}}.server)

